### PR TITLE
Use SDL_GetWindowSize and SDL_WarpMouseInWindow in Windows

### PR DESCRIPTION
**Pull request by q4a**
This is mostly about using SDL_GetWindowSize and SDL_WarpMouseInWindow in Windows
and few include changes.

### CODE REVIEW

**Comment by espkk** [comment]
IIRC this has been already implemented, but then I reverted it back because it worked just awful on windows due to poorly designed sdl implementation...
Was there any changes since then?

**Comment by q4a** [comment]
> IIRC this has been already implemented, but then I reverted it back because it worked just awful on windows due to poorly designed sdl implementation...

Can you find your revert? What was worked awful? `SDL_GetWindowSize` - already used in `SDLWindow::Width()` and `SDLWindow::Height()` - I can use it.
https://github.com/storm-devs/storm-engine/blob/develop/src/libs/window/src/sdl_window.cpp#L44
`SDL_WarpMouseInWindow` - IDK, here is source code:
https://github.com/libsdl-org/SDL/blob/0a48abc860b436387c94e90c13fd0db25772e356/src/events/SDL_mouse.c#L1015

**Comment by q4a** [comment]
I splited PR - this will have only SDL changes.
And this one will have all other - https://github.com/storm-devs/storm-engine/pull/448

**Comment by espkk** [comment]
I'm afraid that exposing `SDL_Window` is not an option to go. It ruins the whole point of `OSWindow` abstraction
Instead, you can add window size query to the `OSWindow` interface which can be acquired via core API

**Comment by q4a** [comment]
@espkk sorry for delay. I rebased PR on top of develop and removed exposing SDL_Window from OSWindow class.

File: src/libs/window/include/os_window.hpp [hunk 1]
```diff
@@ -38,6 +38,8 @@ class OSWindow
     virtual int Width() const = 0;
     //! Current height of window
     virtual int Height() const = 0;
+    //! Current window size
+    virtual void WindowSize(int *width, int *height) const = 0;
```
Comment (espkk) [inline src/libs/window/include/os_window.hpp]: Could you return a tuple or pair or struct instead of args?

Comment (q4a) [reply src/libs/window/include/os_window.hpp]: Changed to struct.

**Review by Hammie — APPROVED** [review]
I also did a local merge and it seems to work well on Windows.

**Pull request merged**